src/trace.ml

```diff
@@ -45,6 +45,46 @@ let create_elf ~executable ~(when_to_snapshot : When_to_snapshot.t) =
   | Magic_trace_or_the_application_terminates, _ | _, Some _ -> return (Ok elf)
 ;;
 
+let evaluate_symbol_selection ~symbol_selection ~elf ~header =
+  let open Deferred.Or_error.Let_syntax in
+  match elf with
```

Comment:
Weakly think this should be written as

```ocaml
let%bind.Deferred.Or_error () =
   match elf with
   | ...
in
```

to decrease nesting.

Reply:
Made that change, I agree it looks better